addons/stock/models/stock_move.py

```diff
@@ -353,11 +353,14 @@ def create(self, vals):
     def write(self, vals):
         # FIXME: pim fix your crap
         if vals.get('product_uom_qty'):
+            for move in self:
+                move.picking_id.message_post_with_view('stock.track_move_template',
+                    values = {'move': move, 'product_uom_qty': vals.get('product_uom_qty')},
+                    subtype_id = self.env.ref('mail.mt_note').id)
```

Comment:
I am afraid we should do this on stock.move.line instead.  When you set initial demand in draft, it should not be logged. When you change lots/location/... on a done stock.move.line, it should also be logged.

Reply:
And it is here that you should inherit in mrp in the case there is an mo related to it.

Reply:
We have to put message on both side ( Move : in case of when we change initial demand from stock move // without stock move lines ) and ( Move line : When picking is done ) untill that we don't have to put log on picking for move line changes.

Reply:
I think you are right on this one.